task/aontimer/src/main.rs

```diff
@@ -0,0 +1,79 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+#![no_main]
+
+use aontimer as aon;
+use drv_riscv_plic_api::*;
+use userlib::*;
+
+task_slot!(INT_CONTROLLER, ext_int_ctrl);
+
+//TODO should be replaced by build time constants
+// see jira https://rivosinc.atlassian.net/browse/SW-530
+const AONTIMER_BARK_NOTIFICATION: u32 = 0x1;
+const AONTIMER_TIME_NOTIFICATION: u32 = 0x2;
+
+const BARK_INT: u32 = 0;
+
+fn sleep_and_listen(timer: &aon::AonTimer) {
+    sys_log!("Sleeping...");
+    let alarm = sys_get_timer().now;
+    sys_set_timer(Some(alarm + 1000), AONTIMER_TIME_NOTIFICATION);
+    loop {
+        let result = sys_recv_closed(
+            &mut [],
+            AONTIMER_BARK_NOTIFICATION | AONTIMER_TIME_NOTIFICATION,
+            TaskId::KERNEL,
+        )
+        .unwrap();
+        if result.operation & AONTIMER_BARK_NOTIFICATION != 0x0 {
+            sys_log!("{}", sys_get_timer().now);
+            //reset the interrupt
+            if timer.bark_cb.is_some() {
+                sys_log!("calling back");
+                (timer.bark_cb.unwrap())();
+            }
+        }
+        if result.operation & AONTIMER_TIME_NOTIFICATION != 0x0 {
+            // Comment the following line of code to recieve the bark, and then be forced reset.
+            return; //You've woken up
+        }
+    }
+}
+
+#[export_name = "main"]
+fn main() -> ! {
+    let int_ctrl = RiscvIntCtrl::from(INT_CONTROLLER.get_task_id());
+
+    int_ctrl.disable_int(BARK_INT).unwrap();
+
+    //TODO change to const read from chip config
+    //see jira https://rivosinc.atlassian.net/browse/SW-431
+    const AONTIMER_BASE: u32 = 0x4047_0000;
+    const AONTIMER_FREQ: u32 = 200_000;
+    sys_log!("Restarted...");
+
+    int_ctrl.enable_int(BARK_INT).unwrap();
+
+    let timer = aon::AonTimer::new(
+        AONTIMER_BASE,
+        AONTIMER_FREQ,
+        2000,
+        5000,
+        Some(|| {
```

Comment:
I noticed other apps have configs within the app.toml. I think we should do the same here.
```
[tasks.aontimer_demo.config]
clock_frequency = ...
bark_threshold = ...
bite_threshold = ...
```
Then we could ensure the bark < bite relationship at compile time.

Reply:
Changed